### PR TITLE
docs: add references to Vite DevServer Proxy documentation

### PR DESCRIPTION
**Pull request by amirhosseinadev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: adev/src/content/tools/cli/serve.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ For example, to divert all calls for `http://localhost:4200/api` to a server run
 1. To run the development server with this proxy configuration, call `ng serve`.
 
 Edit the proxy configuration file to add configuration options; following are some examples.
-For a description of all options, see [webpack DevServer documentation](https://webpack.js.org/configuration/dev-server/#devserverproxy).
+For a detailed description of all options, refer to the [webpack DevServer documentation](https://webpack.js.org/configuration/dev-server/#devserverproxy) when using `@angular-devkit/build-angular:browser`, or the [Vite DevServer documentation](https://vite.dev/config/server-options#server-proxy) when using `@angular-devkit/build-angular:browser-esbuild` and `@angular-devkit/build-angular:application`.
```
Comment (JeanMeche) [inline adev/src/content/tools/cli/serve.md:77]: nit: 

```suggestion
For a detailed description of all options, refer to the [webpack DevServer documentation](https://webpack.js.org/configuration/dev-server/#devserverproxy) when using `@angular-devkit/build-angular:browser`, or the [Vite DevServer documentation](https://vite.dev/config/server-options#server-proxy) when using `@angular-devkit/build-angular:browser-esbuild` or `@angular-devkit/build-angular:application`.
```

**Review by JeanMeche — APPROVED** [review]
Beside the nit, LGTM

**Comment by JeanMeche** [comment]
Beside the nit, could you please reword the commit and drop the scope. There is no need for a scope for documentation changes. Thank you.

**Comment by alan-agius4** [comment]
@yaamisv, unfortunately, we cannot merge your contribution unless you sign the CLA.

**Comment by yaamisv** [comment]
> @yaamisv, unfortunately, we cannot merge your contribution unless you sign the CLA.

thanks for your message. I've edited PR title, commit message and signed the CLA.

File: adev/src/content/tools/cli/serve.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Serving Angular apps for development
+    # Serving Angular apps for development
```
Comment (JeanMeche) [inline adev/src/content/tools/cli/serve.md:1]: Can you make sure to remove the leading spaces.

**Review by alan-agius4 — APPROVED** [review]

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit ba2a5d5966aa9b68bab63c042666b3a28c1905d6.

The changes were merged into the following branches: main, 19.0.x

**Pull request closed without merging**